python/cudf/cudf/utils/dtypes.py

```diff
@@ -444,3 +444,12 @@ def get_time_unit(obj):
 
     time_unit, _ = np.datetime_data(obj.dtype)
     return time_unit
+
+
+def _get_nan_for_dtype(dtype):
+    if pd.api.types.is_datetime64_dtype(
+        dtype
+    ) or pd.api.types.is_timedelta64_dtype(dtype):
+        return dtype.type("nat")
```

Comment:
Should type this with the units of the `datetime64` type that's passed in.

Reply:
Done 👍